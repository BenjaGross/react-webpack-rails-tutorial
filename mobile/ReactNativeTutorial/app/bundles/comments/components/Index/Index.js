// @flow
import React from 'react';
import { View } from 'react-native';
import List from './List/List';
import Footer from './Footer/Footer';
import withIndexProps from '../../hocs/withIndexProps';

import styles from './IndexStyle';

type PropsType = {

}

class Index extends React.Component {

  props: PropsType;

  componentDidMount() {
    this.props.actions.fetch();
  }

  render() {
    return (
      <View style={styles.container}>
        <List {...this.props} />
        <Footer />
      </View>
    );
  }
}

export default withIndexProps(Index);
