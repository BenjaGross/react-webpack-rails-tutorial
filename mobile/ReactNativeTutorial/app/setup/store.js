import { applyMiddleware, createStore } from 'redux';
import effectsMiddleware from 'redux-thunk-effects';
import loggerMiddleware from './loggerMiddleware';
import appReducer, { initialState } from '../reducers';

const middlewares = __DEV__ ? [effectsMiddleware, loggerMiddleware] : [effectsMiddleware];
export default createStore(appReducer, initialState, applyMiddleware(...middlewares));
