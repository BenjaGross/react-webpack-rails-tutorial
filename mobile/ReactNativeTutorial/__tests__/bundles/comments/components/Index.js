import React from 'react';
import store from 'ReactNativeTutorial/app/setup/store';
import Index from 'ReactNativeTutorial/app/bundles/comments/components/Index/Index';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const response = {
    comments: [
      { id: 1, author: 'Alexey', text: 'Just some random comment' },
      { id: 2, author: 'Justin', text: 'Another random comment' },
    ],
  };
  fetch.mockResponseSuccess(response);
  const tree = renderer.create(
    <Index store={store} />
  );
  expect(tree).toMatchSnapshot();
});
