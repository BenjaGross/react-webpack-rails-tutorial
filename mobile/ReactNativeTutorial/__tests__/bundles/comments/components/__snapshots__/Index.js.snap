exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    refreshControl={
      <RefreshControl
        onRefresh={[Function]}
        refreshing={true}
        tintColor="#4641B5" />
    }
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    style={Object {}} />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EFEFF2",
        "borderTopColor": "#D2D2D2",
        "borderTopWidth": 1,
        "height": 70,
        "justifyContent": "center",
      }
    }>
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      style={
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f50",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="#f50">
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "Material Icons",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
          ]
        }>
        î……
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
